memcached.c

```diff
@@ -5750,321 +6019,162 @@ static void process_command(conn *c, char *command) {
     }
 
     ntokens = tokenize_command(command, tokens, MAX_TOKENS);
-    if (ntokens >= 3 &&
-        ((strcmp(tokens[COMMAND_TOKEN].value, "get") == 0) ||
-         (strcmp(tokens[COMMAND_TOKEN].value, "bget") == 0))) {
 
-        process_get_command(c, tokens, ntokens, false, false);
+    // Meta commands are all 2-char in length.
+    if (tokens[COMMAND_TOKEN].length == 2 &&
+            tokens[COMMAND_TOKEN].value[0] == 'm') {
+        switch (tokens[COMMAND_TOKEN].value[1]) {
+            case 'g':
+                process_mget_command(c, tokens, ntokens);
+                break;
+            case 's':
+                process_mset_command(c, tokens, ntokens);
+                break;
+            case 'd':
+                process_mdelete_command(c, tokens, ntokens);
+                break;
+            case 'n':
+                out_string(c, "MN");
+                // mn command forces immediate writeback flush.
+                conn_set_state(c, conn_mwrite);
+                break;
+            case 'a':
+                process_marithmetic_command(c, tokens, ntokens);
+                break;
+            case 'e':
+                process_meta_command(c, tokens, ntokens);
+                break;
+        }
+    } else if (tokens[COMMAND_TOKEN].value[0] == 'g') {
```

Comment:
if input string is just a space(" "), segfault occurs here.
Please check it.

Reply:
good catch! I left out the default: there. missing coverage in t/metaget.t